### PR TITLE
feat(poe-v2): scaffolding for PoE v2 on event queries, full support in retention queries

### PR DESCRIPTION
**Pull request by yakkomajuri**
#14792 and the general top-down approach (instead of the previous bottom-up) really made things much easier.

This PR gets the scaffolding in `event_query` for PoE v2 as well as adds full support in retention queries

### CODE REVIEW

File: ee/clickhouse/queries/enterprise_cohort_query.py [hunk 1]
```diff
@@ -298,7 +298,7 @@ def _get_sequence_query(self) -> Tuple[str, Dict[str, Any], str]:
 
         new_query = f"""
         SELECT {", ".join(_inner_fields)} FROM events AS {self.EVENT_TABLE_ALIAS}
-        {self._get_distinct_id_query()}
+        {self._get_person_ids_query()}
```
Comment (yakkomajuri) [inline ee/clickhouse/queries/enterprise_cohort_query.py:301]: thought the name was more appropriate, even before the changes

File: ee/clickhouse/queries/test/__snapshots__/test_retention.ambr [hunk 2]
```diff
@@ -656,6 +656,191 @@
...
+---
+# name: TestClickhouseRetention.test_groups_filtering_person_on_events_v2.1
+  '
+  WITH actor_query AS
+    (WITH 'Day' as period,
+          NULL as breakdown_values_filter,
+          NULL as selected_interval,
+          returning_event_query as
+       (SELECT toStartOfDay(toTimeZone(toDateTime(e.timestamp, 'UTC'), 'UTC')) AS event_date,
+               if(notEmpty(overrides.person_id), overrides.person_id, e.person_id) as target
```
Comment (yakkomajuri) [inline ee/clickhouse/queries/test/__snapshots__/test_retention.ambr:680]: note this

File: ee/clickhouse/queries/test/__snapshots__/test_retention.ambr [hunk 3]
```diff
@@ -656,6 +656,191 @@
...
+        FROM events e
+        LEFT OUTER JOIN
```
Comment (yakkomajuri) [inline ee/clickhouse/queries/test/__snapshots__/test_retention.ambr:682]: and this

File: posthog/test/base.py [hunk 4]
```diff
@@ -763,3 +763,24 @@ def _create_insight(
     insight = Insight.objects.create(team=team, filters=insight_filters)
     dashboard_tile = DashboardTile.objects.create(dashboard=dashboard, insight=insight)
     return insight, dashboard, dashboard_tile
+
+
+def create_person_id_override_by_distinct_id(distinct_id_from: str, distinct_id_to: str, team_id: int):
```
Comment (yakkomajuri) [inline posthog/test/base.py:768]: reasonably rough for now but does the job nicely. as I go along I might set up some decorators and nicer helpers for testing

File: ee/clickhouse/queries/test/__snapshots__/test_retention.ambr (hunk 3 above)
Comment (neilkakkar) [reply ee/clickhouse/queries/test/__snapshots__/test_retention.ambr:682]: Q: Why did we have to do the whole person override join here? Seems like it's a groups query, which just depends on events, not person IDs?

File: posthog/queries/test/__snapshots__/test_retention.ambr [hunk 5]
```diff
@@ -70,6 +70,89 @@
            intervals_from_base
   '
 ---
+# name: TestFOSSRetention.test_month_interval_with_person_on_events_v2
+  '
+  
+  SELECT distinct_id,
```
Comment (neilkakkar) [inline posthog/queries/test/__snapshots__/test_retention.ambr:76]: where is this used? 🤔

Comment (neilkakkar) [reply posthog/queries/test/__snapshots__/test_retention.ambr:76]: I see at the end, it's test scaffolding 👍

**Review by neilkakkar — APPROVED** [review]

File: posthog/queries/retention/retention_events_query.py [hunk 6]
```diff
@@ -191,17 +189,20 @@ def get_timestamp_field(self) -> str:
             return f"{self._trunc_func}(toTimeZone(toDateTime({self.EVENT_TABLE_ALIAS}.timestamp, 'UTC'), %(timezone)s)) AS event_date"
 
     def _determine_should_join_distinct_ids(self) -> None:
-        if (
+        non_person_id_aggregation = (
             self._filter.aggregation_group_type_index is not None or self._aggregate_users_by_distinct_id
-        ) and not self._column_optimizer.is_using_cohort_propertes:
+        )
+        is_using_cohort_propertes = self._column_optimizer.is_using_cohort_propertes
+        if self._person_on_events_mode == PersonOnEventsMode.V1_ENABLED or (
```
Comment (tiina303) [inline posthog/queries/retention/retention_events_query.py:196]: shouldn't this be the case for PoEv2 as well (we don't want to join distinct_ids then either, do we?)

**Pull request merged**